### ISSUE TITLE
On using the table shortcut key, ‘Ctrl+Alt+Arrow’ keys for navigation in the grid cells, NVDA announces as ‘not in a table cell’. 

### ISSUE THREAD

**Issue by msftedad** [nvaccess/nvda#13476]
### Steps to reproduce:
1.	Turn on the screen reader NVDA.
2.     Go to this URL https://www.w3.org/TR/wai-aria-practices-1.1/examples/table/table.html
3.	Turn on the scan mode of NVDA using ‘insert+spacebar’.
4.	Press table shortcut key ‘T’ in scan mode to navigate to the table named ‘Students currently enrolled in WAI-ARIA 101 for the coming semester’.
5.	Once the screen reader focus reaches to the table then use the table shortcut ‘Ctrl+Alt+Arrow’ keys to verify whether NVDA is navigating to the table cells or not?

### Actual behavior:
On using the table shortcut key, ‘Ctrl+Alt+Arrow’ keys for navigation in the grid cells, NVDA announces as ‘not in a table cell’. But on pressing ‘Ctrl+Down arrow’ key and then on using ‘Ctrl+Alt+Arrow’ keys, NVDA is navigating through the grid cells properly.

### Expected behavior:
On using the table shortcut key, ‘Ctrl+Alt+Arrow’ keys , NVDA should properly be able to navigate through the table cells and it should not announce like ‘not in a table cell’.

#### NVDA version:
NVDA Version 2021.3.3

#### Windows version:
Microsoft Windows Version 21H2 (OS Build 22000.493)

#### Name and version of other software in use when reproducing the issue:
Microsoft Edge Version 100.0.1163.1 (Official build) dev (64-bit)

#### Does the issue still occur after restarting your computer?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No

#### If NVDA add-ons are disabled, is your problem still occurring?
Yes

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by XLTechie** [comment]
I believe that this is intended behavior.

The table isn't just made up of its rows and columns, but also a description div 
which precedes them.

If the "t" command didn't take the user to the start of the table, including its 
description/caption/etc., the user could (in this case would) lose important 
information.

For example, if you delete the div with id:
"students_table_desc"
The table navigation speaks like I think you desire, but then only the 
aria-label of "students" is spoken when you reach the table, and not the full 
table description.

Narrator, in fact, does exactly that: skipping the description entirely, unless 
you navigate up from the table. In that case, you have no idea that there is 
even a description there, unless you look for it.

Interestingly, testing this with Narrator in a Chromium browser vs. Firefox, 
really shows some of Narrator's table handling shortcomings in Firefox.

IMO, the only viable way for NVDA to do what you want, and still give as much 
useful information as it does now, is to:

1. Skip browsing to the description div, and instead browse directly to the 
first cell; and

2. Speak the description immediately after the table label, so anyone inclined 
to interrupt speech before the row/column count and first headings are given, 
will still hear (at least the start of) the description.

Which would be a reasonable feature request I suppose.

**Comment by XLTechie** [comment]
I would imagine so, although "table has description" takes almost as long to say 
as many table descriptions do. That may frustrate users. I know it would 
frustrate me.

I'd sooner have the description spoken before the column/row count as I 
mentioned.

**Comment by michaelDCurran** [comment]
I'm leaning towards changing t to jump to the first table cell. However, it does then mean the author would have to correctly set aria-describedby on the table pointing to that initial div.
Would people agree with this?

**Comment by msftedad** [comment]
From our perspective, we would like this to be a feature.

**Comment by XLTechie** [comment]
@michaelDCurran wrote:
> I'm leaning towards changing t to jump to the first table cell. However, it does then mean the author would have to correctly set aria-describedby on the table pointing to that initial div.

Would you consider that to have been done correctly in the W3C example linked in the STR?

If so: what would be NVDA's behavior in relation to the description? As it stands, going to that first cell does not cause the description to be read, or the existence of a description to be indicated.

**Comment by michaelDCurran** [comment]
Actually, we would have to also ensure that NVDA does report the table's 
description when jumping with quicknav. I don't believe that happens 
currently.

I can understand why we previously did not want this happening as each 
time you'd always land on that exact text anyway.

**Comment by XLTechie** [comment]
In that case, I too would support this change.

**Comment by Adriani90** [comment]
I would support the proposed change as well.

**Comment by gabe-piltzer-zendesk** [comment]
In the example linked in the steps to reproduce, the table is made of divs. My table uses the `table` and `caption` elements. NVDA correctly reads the number of rows and columns, as well as the caption. But when trying to traverse the table it's just "not a table cell" the entire way. I couldn't find any permutation that included both of those elements that would work. I tested with NVDA 2023.1 in Firefox 113. I repeated the same test using the Narrator and had no issues. 

Is it expected that the DOM structure from this screenshot would work in NVDA?

![Screenshot 2023-05-25 at 8 00 44 PM](https://github.com/nvaccess/nvda/assets/91081358/aa99fbcb-c0b8-4f6c-857f-43c1ac76cf8a)

**Comment by gabe-piltzer-zendesk** [comment]
NM my previous comment. This appears to be working in our production environment. Thanks

**Comment by SaschaCowley** [comment]
@gabe-piltzer-zendesk per our [documentation on creating new issues on the NVDA project](https://github.com/nvaccess/nvda/blob/master/projectDocs/issues/githubIssueTemplateExplanationAndExamples.md#screenshots), please keep in mind that many of the developers of NVDA are blind and will greatly appreciate images being described in text. Thanks :)
